scripts/o3de/ExportScripts/export_source_android.py

```diff
@@ -0,0 +1,383 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+import o3de.command_utils as command_utils
+import o3de.export_project as exp
+import o3de.manifest as manifest
+
+from o3de import android, android_support
+
+
+import argparse
+import logging
+import pathlib
+import platform
+import sys
+import time
+
+from typing import List
+
+def supply_android_default_global_configuration(api:int = 31,
+                                        ndk_version:str = '25*',
+                                        gradle_min_version:str = '8.1.0',
+                                        auto_store_pwd:str|None = None,
+                                        auto_key_pwd:str|None = None):
+    is_pc = exp.get_default_asset_platform() == 'pc'
+    android_sdk_home_env_var = '%ANDROID_SDK_HOME%' if is_pc else '$ANDROID_SDK_HOME'
+    android_signing_config_keystore_file_env_var = '%ANDROID_SIGNING_CONFIG_KEYSTORE_FILE%' if is_pc else '$ANDROID_SIGNING_CONFIG_KEYSTORE_FILE' 
+    android_signing_config_key_alias_env_var = '%ANDROID_SIGNING_CONFIG_KEY_ALIAS%' if is_pc else '$ANDROID_SIGNING_CONFIG_KEY_ALIAS'
+
+    android_global_config, _ = android.get_android_config(is_global=True)
+    android_global_config.set_config_value(key=android_support.SETTINGS_SDK_ROOT.key, value=android_sdk_home_env_var, validate_value=False)
+    android_global_config.set_config_value(key=android_support.SETTINGS_PLATFORM_SDK_API.key, value=api, validate_value=False)
+    android_global_config.set_config_value(key=android_support.SETTINGS_NDK_VERSION.key, value=ndk_version, validate_value=False)
+    android_global_config.set_config_value(key=android_support.SETTINGS_GRADLE_PLUGIN_VERSION.key, value=gradle_min_version, validate_value=False)
+    android_global_config.set_config_value(key=android_support.SETTINGS_SIGNING_STORE_FILE.key, value=android_signing_config_keystore_file_env_var, validate_value=False)
+    android_global_config.set_config_value(key=android_support.SETTINGS_SIGNING_KEY_ALIAS.key, value=android_signing_config_key_alias_env_var, validate_value=False)
+
+    if not auto_store_pwd:
+        android_global_config.set_password(android_support.SETTINGS_SIGNING_STORE_PASSWORD.key)
+    else:
+        android_global_config.set_config_value(key=android_support.SETTINGS_SIGNING_STORE_PASSWORD.key, value=auto_store_pwd, validate_value=False)
+    
+    if not auto_key_pwd:
+        android_global_config.set_password(android_support.SETTINGS_SIGNING_KEY_PASSWORD.key)
+    else:
+        android_global_config.set_config_value(key=android_support.SETTINGS_SIGNING_KEY_PASSWORD.key, value=auto_key_pwd, validate_value=False)
+
+def deloy_to_android_device(target_android_project_path: pathlib.Path,
```

Comment:
This should be moved to the o3de.android_support.py file since its more generalized

Reply:
I don't see this method used at all here.

Reply:
Going ahead and moving the function there. Currently the deploy functionality is implemented using `process_command`, so I will be migrating it with that code in tact (but will include error handling for command failures). But I am currently looking in `android.py` and `android_support.py` to see if we already have a pre-existing API for handling deployments.